script/gen-theme.js

```diff
@@ -0,0 +1,172 @@
+    "linear-gradient(#111111, #000000)",
+  ],
+  "button-inset-background": [
+    "linear-gradient(#00000022, #00000011)",
+    "linear-gradient(#222222, #333333)",
+    "linear-gradient(#222222, #333333)",
+  ],
+  "button-shadow": [
+    "0 0.33px 0 1px #ffffff26, 0 4px 8px -3px #00000026, 0 0.4px 0 0.8px #00000040",
+    "0 -0.33px 0 1px #ffffff33, 0 4px 8px -3px #00000026, 0 0.4px 0 0.8px #00000040",
+    "0 -0.33px 0 1px #ffffff33, 0 4px 8px -3px #00000026, 0 0.4px 0 0.8px #00000040",
+  ],
+  "button-hover-shadow": [
+    "0 0.33px 0 1px #ffffff26, 0 6px 8px -4px #00000033, 0 0.4px 0 0.8px #00000059",
+    "0 -0.33px 0 1px #ffffff55, 0 6px 8px -4px #00000033, 0 0.4px 0 0.8px #00000059",
+    "0 -0.33px 0 1px #ffffff55, 0 6px 8px -4px #00000033, 0 0.4px 0 0.8px #00000059",
+  ],
+  "button-active-shadow": [
+    "0 0.33px 0 1px #ffffff26, 0 4px 6px -3px #00000026, 0 0.4px 0 0.8px #00000073",
+    "0 -0.33px 0 1px #ffffff44, 0 4px 6px -3px #00000026, 0 0.4px 0 0.8px #00000073",
+    "0 -0.33px 0 1px #ffffff44, 0 4px 6px -3px #00000026, 0 0.4px 0 0.8px #00000073",
+  ],
+  "button-inset-shadow": [
+    "0 0.33px 0 1px #ffffff26 inset, 0 4px 8px -3px #00000026 inset, 0 0.4px 0 0.8px #00000040 inset",
+    "0 -0.33px 0 1px #ffffff44 inset, 0 4px 8px -3px #00000026 inset, 0 0.4px 0 0.8px #00000040 inset",
+    "0 -0.33px 0 1px #ffffff44 inset, 0 4px 8px -3px #00000026 inset, 0 0.4px 0 0.8px #00000040 inset",
+  ],
+  /* Shadows */
+  "heavy-3d-box-shadow": [
+    "0 0.33px 0 1px #ffffff26, 0 8px 12px -6px #0000004d, 0 0.5px 0 1px #00000033",
+    "0 -0.33px 0 1px #ffffff26, 0 8px 12px -6px #0000004d, 0 0.5px 0 1px #00000033",
+    "0 -0.33px 0 1px #ffffff26, 0 8px 12px -6px #0000004d, 0 0.5px 0 1px #00000033",
+  ],
+  "text-input-shadow": [
+    "0 0.33px 0 1px #ffffff26, 0 6px 9px -4px #0000001a inset, 0 0.4px 0 0.8px #0000001a inset",
+    "0 -0.33px 0 1px #ffffff26, 0 6px 9px -4px #0000001a inset, 0 0.4px 0 0.8px #0000001a inset",
+    "0 -0.33px 0 1px #ffffff26, 0 6px 9px -4px #0000001a inset, 0 0.4px 0 0.8px #0000001a inset",
+  ],
+  "dropdown-shadow": [
+    "0 0.33px 0 1px #ffffff26, 0 12px 12px -4px #00000033, 0 0.4px 0 0.8px #00000059",
+    "0 -0.33px 0 1px #ffffff55, 0 12px 12px -4px #00000033, 0 0.4px 0 0.8px #00000059",
+    "0 -0.33px 0 1px #ffffff55, 0 12px 12px -4px #00000033, 0 0.4px 0 0.8px #00000059",
+  ],
+};
+
+var childVariables = {
+  "log-color": [
+    "var(--light-log-color)",
+    "var(--dark-log-color)",
+    "var(--dark-log-color)",
+  ],
+  "log-bg-color": [
+    "var(--light-log-bg-color)",
+    "var(--dark-log-bg-color)",
+    "var(--dark-log-bg-color)",
+  ],
+  "log-hover-color": [
+    "var(--light-log-hover-color)",
+    "var(--dark-log-hover-color)",
+    "var(--dark-log-hover-color)",
+  ],
+  "log-bg-hover-color": [
+    "var(--light-log-bg-hover-color)",
+    "var(--dark-log-bg-hover-color)",
+    "var(--dark-log-bg-hover-color)",
+  ],
+};
+
+function themeDefinition(theme) {
```

Comment:
nit: prefer arrow functions unless you need mutual recursion or other obscure reasons for using `function`